core/src/main/scala/kafka/cluster/LogInfo.scala

```diff
@@ -0,0 +1,146 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.cluster
+
+import kafka.common.KafkaException
+import kafka.log.{Log, LogOffsetSnapshot}
+import kafka.server.{LogOffsetMetadata, OffsetAndEpoch}
+import kafka.utils.Logging
+import org.apache.kafka.common.errors.OffsetOutOfRangeException
+import org.apache.kafka.common.TopicPartition
+
+/**
+ * Class to hold information about local partition log. A Partition
+ * can have at most two logs:
+ * <ol>
+ * <li> Log where all current active write are going (called "local replica").
+ * <li> If an "AlterLogDirs" command is issued to move log directory,
+ *      then the Partition gets another log (called "future replica").
+ * </ol>
+ *
+ * Having two logs at same time is a temporary state and as soon as log directory
+ * is successfully moved a Partition will again have only one Log.
+ */
+class LogInfo private(val brokerId: Int,
```

Comment:
Need a suggestion for name of this class.  The class is a wrapper around Log and contains method used to be in Replica class before (managing HW and offsets)

Reply:
Fixed. Removed the class and pushed high watermark logic to Log.